### PR TITLE
nixos/locate: rename option `locate` -> `package`

### PR DESCRIPTION
**Pull request by bobvanderlinden**
## Description of changes

Currently `services.locate.locate` is an option to define the package of the locate binary. This is quite confusing when regularly working with NixOS modules/services, as most of them have a `package` option. This PR attempts to rename `services.locate.locate` to `services.locate.package`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 1]
```diff
@@ -261,6 +261,8 @@
 
 - Setting `nixpkgs.config` options while providing an external `pkgs` instance will now raise an error instead of silently ignoring the options. NixOS modules no longer set `nixpkgs.config` to accomodate this. This specifically affects `services.locate`, `services.xserver.displayManager.lightdm.greeters.tiny` and `programs.firefox` NixOS modules. No manual intervention should be required in most cases, however, configurations relying on those modules affecting packages outside the system environment should switch to explicit overlays.
 
+- The `services.locate.locate` has been renamed to `services.locate.package` to align with the convention of other services.
+
```
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2311.section.md:265]: I don't think we need this. There are no breaking changes and the warning with the renaming should be enough.

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM

File: nixos/doc/manual/release-notes/rl-2311.section.md (hunk 1 above)
Comment (bobvanderlinden) [reply nixos/doc/manual/release-notes/rl-2311.section.md]: Thanks for the quick review! I've removed the release note :+1:

**Review by bjornfor — APPROVED** [review]

**Review by jian-lin — COMMENTED** [review]
LGTM

Could you change the commit message and the PR title to `nixos/locate: ...`?

**Comment by bobvanderlinden** [comment]
> Could you change the commit message and the PR title to `nixos/locate: ...`?

Done :+1:

**Review by jian-lin — APPROVED** [review]

**Comment by jian-lin** [comment]
@ofborg test locate

**Pull request merged**